aspnet/fundamentals/app-state.rst

```diff
@@ -1,8 +1,252 @@
-.. include:: /../common/stub-topic.txt
+Managing Application State
+==========================
+By `Steve Smith`_
 
-|stub-icon| Managing Application State
-======================================
+In ASP.NET 5, application state can be managed in a variety of ways, depending on when and how the state is to be retrieved. This article provides a brief overview of several options, and focuses on installing and configuring Session state support in ASP.NET 5 applications.
+
+.. contents:: Sections
+	:local:
+	:depth: 1
+
+`Download sample from GitHub <https://github.com/aspnet/docs/aspnet/fundamentals/app-state/sample>`_. 
+
+Application State Options
+-------------------------
+`Application state` refers to any data that is used to represent the current representation of the application. This includes both global and user-specific data. Previous versions of ASP.NET (and even ASP) have had built-in support for global ``Application`` and ``Session`` state stores, as well as a variety of other options. 
+
+.. note:: The ``Application`` store had the same characteristics as the ASP.NET ``Cache``, with fewer capabilities. In ASP.NET 5, ``Application`` no longer exists; applications written for previous versions of ASP.NET that are migrating to ASP.NET 5 replace ``Application`` with a :doc:`caching` implementation.
+
+Application developers are free to use different state storage providers depending on a variety of factors:
+
+- How long does the data need to persist?
+- How large is the data?
+- What format is the data?
+- Can it be serialized?
+- How sensitive was the data? Could it be stored on the client?
+
+Based on answers to these questions, application state in ASP.NET 5 apps can be stored or managed in a variety of ways.
+
+HttpContext.Items
+^^^^^^^^^^^^^^^^^
+The ``Items`` collection is the best location to store data that is only needed while processing a given request. Its contents are disposed of with each request. It is best used as a means of communicating between components or middleware that operate at different points in time during a request, and have no direct relationship with one another through which to pass parameters or return values. See `Working with HttpContext.Items`_, below.
+
+Querystring and Post
+^^^^^^^^^^^^^^^^^^^^
+State from one request can be provided to another request by adding values to the new request's querystring or by POSTing the data. This requires that the data be sent to the client and then sent back to the server, so it is best for small amounts of data. Querystrings are especially useful for capturing state in a persistent manner, allowing links with embedded state to be created and sent via email or social networks, for use potentially far into the future.
```

Comment:
Need to mention XSRF risks here. This kind of state should be not be used for sensitive data.